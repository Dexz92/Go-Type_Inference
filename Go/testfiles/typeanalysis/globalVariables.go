var a = 5;
var a2 = true + 3;
var c = 5.9;
var f float64 = 2;

func main() {
  var b int = a;
  var d bool = a + c;
  var g int = f;
}
