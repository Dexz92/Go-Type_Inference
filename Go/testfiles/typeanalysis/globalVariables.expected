Error at line 2: Invalid operation: true + 3 (mismatched types: bool and int)
Error at line 4: Trying to assign int to global float variable
Error at line 8: Invalid operation: a + c (mismatched types: int and float)
Error at line 9: Trying to assign float to int variable
